controllers/users.js

```diff
@@ -462,6 +464,46 @@ const filterUsers = async (req, res) => {
   }
 };
 
+// one time script function to perform the migration - adding github_user_id field to the document
+const migrate = async (req, res) => {
+  try {
+    // Fetch user data from GitHub API for each document in the users collection
+    // divided by 500 because firestore api guarantee that we can process in batch of 500.
+    const usersSnapshot = await firestore.collection("users").get();
+    const batchCount = Math.ceil(usersSnapshot.docs.length / 500);
+    // Create batch write operations for each batch of documents
+    for (let i = 0; i < batchCount; i++) {
+      const batchDocs = usersSnapshot.docs.slice(i * 500, (i + 1) * 500);
+      const batchWrite = firestore.batch();
+      const batchWrites = [];
+      for (const userDoc of batchDocs) {
+        const githubUsername = userDoc.data().github_id;
+        batchWrite.update(userDoc.ref, { github_user_id: null });
+        batchWrites.push(
+          axios
+            .get(`https://api.github.com/users/${githubUsername}`)
+            .then((response) => {
+              const githubUserId = response.data.id;
+              batchWrite.update(userDoc.ref, { github_user_id: `${githubUserId}` });
+            })
+            .catch((error) => {
+              logger.error(error);
+            })
+        );
+      }
+      await Promise.all(batchWrites);
+      await batchWrite.commit();
+    }
+
+    return res.send({
```

Comment:
In the response, the `statusCode` field will have a `201` value while the response status code will be `200`. It seems confusing
To avoid please use `res.status(200).json({})`